packages/eslint-plugin/src/rules/no-empty-object-type.ts

```diff
@@ -0,0 +1,45 @@
+import type { TSESLint } from '@typescript-eslint/utils';
+
+import { createRule } from '../util';
+
+export default createRule({
+  name: 'no-empty-object-type',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow accidentally using the "empty object" type',
+      recommended: 'recommended',
+    },
+    hasSuggestions: true,
+    messages: {
+      banEmptyObjectType: [
+        'The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.',
+        "- If that's what you want, disable this lint rule with an inline comment or in your ESLint config.",
+        '- If you want a type meaning "any object", you probably want `object` instead.',
+        '- If you want a type meaning "any value", you probably want `unknown` instead.',
+        '- If you want a type meaning "empty object", you probably want `Record<string, never>` instead.',
+      ].join('\n'),
+      replaceEmptyObjectType: 'Replace `{}` with `{{replacement}}`.',
```

Comment:
Do we want to use the dot points above as the suggestion fixer messages for more clarity?

Reply:
Ehhh not really IMO. They'll have to go through the big report message anyway to get here.